### PR TITLE
Fix docstrings

### PR DESCRIPTION
**Pull request by rekmarks**
- Fixes all docstrings in `index.js`
- Felt like I touched hallowed MetaMask history

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -386,13 +415,26 @@ class KeyringController extends EventEmitter {
...
     return keyring.getAppKeyAddress(address, origin)
   }
 
+  /**
+   * TODO: what exactly does this do?
+   *
+   * @param {string} _address - The Ethereum address for the app key.
+   * @param {string} origin - The origin for the app key.
+   * @returns {string} The app key address. TODO: wut?
+   */
```
Comment (rekmarks) [inline index.js]: Not totally sure what's going on here. Can someone please add a suggestion?

Comment (Gudahtt) [reply index.js]: It looks like this exports a an app key (e.g. the private key portion of it)

See here: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1775.md#private-app-key-generation-algorithm-1

Comment (rekmarks) [reply index.js]: Got it, got it. Fixed.

File: index.js [hunk 2]
```diff
@@ -44,47 +42,52 @@ class KeyringController extends EventEmitter {
...
-  //
-  // returns Promise( @object state )
-  //
-  // Destroys any old encrypted storage,
-  // creates a new encrypted store with the given password,
-  // creates a new HD wallet from the given seed with 1 account.
+  /**
+   * CreateNewVaultAndRestore
+   *
+   * @returns {Promise<Object>} A Promise that resolves to the state.
```
Comment (Gudahtt) [inline index.js]: Nit: elsewhere the `@returns` declarations were moved to the bottom of the doc comment I noticed

File: index.js [hunk 3]
```diff
@@ -44,47 +42,52 @@ class KeyringController extends EventEmitter {
...
+   *
+   * Destroys any old encrypted storage,
+   * creates a new encrypted store with the given password,
+   * creates a new HD wallet from the given seed with 1 account.
```
Comment (Gudahtt) [inline index.js]: Should there be a newline between description and parameters? 🤔 We don't seem to be consistent about this

File: index.js [hunk 4]
```diff
@@ -249,9 +263,9 @@ class KeyringController extends EventEmitter {
   }
 
   // Export Account
```
Comment (Gudahtt) [inline index.js]: Missed this one?

File: index.js [hunk 5]
```diff
@@ -320,12 +335,14 @@ class KeyringController extends EventEmitter {
...
-  // returns Promise(@buffer rawSig)
-  //
-  // Attempts to sign the provided @object msgParams.
+  /**
+   * Sign Message
+   *
+   * Attempts to sign the provided message parameters.
+   *
+   * @param {Object} msgParams - The message parameters to sign.
+   * @returns {Promise<Buffer>} The raw signature, if successful.
```
Comment (Gudahtt) [inline index.js]: Nit: is there a reason the `, if successful` was added here, but not on other return descriptions? (e.g. such as the one for `signPersonalMessage`. It is sorta implied when returning a Promise that the thing-in-the-promise won't be returned if the Promise is rejected.

File: index.js [hunk 6]
```diff
@@ -309,7 +324,7 @@ class KeyringController extends EventEmitter {
 
   // SIGNING METHODS
```
Comment (Gudahtt) [inline index.js]: Another missed one?

File: index.js (hunk 3 above)
Comment (rekmarks) [reply index.js]: I'm not sure, I think it's more readable. Will add.

Comment (Gudahtt) [reply index.js]: I think I prefer it as well

File: index.js (hunk 5 above)
Comment (rekmarks) [reply index.js]: I was going back and forth re: this. I'm now of the School of Implied Promise Failure.

**Pull request merged**